zulipterminal/api_types.py

```diff
@@ -167,6 +167,37 @@ class ReactionEvent(TypedDict):
     message_id: int
 
 
+class RealmUserEventPerson(TypedDict, total=False):
+    user_id: int
+
+    full_name: str
+
+    avatar_url: str
+    avatar_source: str
+    avatar_url_medium: str
+    avatar_version: int
+
+    # NOTE: This field will be removed in future as it is redundant with the user_id
+    # email: str
+    timezone: str
+
+    role: int
+
+    email: str
```

Comment:
In comparing the fields that can be updated, this one stands out as different from the remaining one field possible (custom_profile_field), unless I'm missing another use of email?

Reply:
The `email` field does not exist under `RealmUserEventPerson`, rather we get `new_email` from the API. However, in `model.initial_data["realm_users"]`, we use `email` and not `new_email`. Hence, in the updated code, where we now have a conditional for replacing the `new_email` key with the `email` key, mypy returns an error.

Reply:
Ah, I see your point. Can we avoid that by moving the conditional into the loop where we've found the user-id?

Reply:
OK, let me just read the code again, I may just merge these first two commits roughly as they are.